packages/pulseaudio/module-sles-sink.c

```diff
@@ -239,99 +290,81 @@ static void process_rewind(pa_sink *s) {
-        goto fail;
-    }
+    pa_sink_new_data_init(&sink_data);
+    sink_data.driver = __FILE__;
+    sink_data.module = m;
 
     ss = m->core->default_sample_spec;
-    pa_channel_map_init_stereo(&map);
-
-    switch (ss.format) {
-    case PA_SAMPLE_S16LE:
-    case PA_SAMPLE_S24LE:
-    case PA_SAMPLE_S32LE:
-    case PA_SAMPLE_FLOAT32LE:
-        break;
-    default:
-        pa_log("Sample format not supported");
+    if (pa_modargs_get_sample_rate(ma, &ss.rate) < 0) {
+        pa_log("Invalid rate specification.");
         goto fail;
     }
-    pa_modargs_get_sample_rate(ma, &ss.rate);
+
+    pa_channel_map_init_stereo(&map);
     ss.channels = map.channels;
 
-    if (pa_init_sles_player(u, &ss) < 0)
+    if (pa_init_sles_player(u, &ss))
         goto fail;
 
-    pa_sink_new_data_init(&data);
-    data.driver = __FILE__;
-    data.module = m;
-    pa_sink_new_data_set_name(&data, pa_modargs_get_value(ma, "sink_name", DEFAULT_SINK_NAME));
-    pa_sink_new_data_set_sample_spec(&data, &ss);
-    pa_sink_new_data_set_channel_map(&data, &map);
-    pa_proplist_sets(data.proplist, PA_PROP_DEVICE_DESCRIPTION, _("OpenSL ES Output"));
-    pa_proplist_sets(data.proplist, PA_PROP_DEVICE_CLASS, "abstract");
-
-    if (pa_modargs_get_proplist(ma, "sink_properties", data.proplist, PA_UPDATE_REPLACE) < 0) {
-        pa_log("Invalid properties");
-        pa_sink_new_data_done(&data);
+    pa_sink_new_data_set_name(&sink_data, pa_modargs_get_value(ma, "sink_name", DEFAULT_SINK_NAME));
+    pa_sink_new_data_set_sample_spec(&sink_data, &ss);
+    pa_sink_new_data_set_channel_map(&sink_data, &map);
+    pa_proplist_sets(sink_data.proplist, PA_PROP_DEVICE_DESCRIPTION, "OpenSL ES Output");
+    pa_proplist_sets(sink_data.proplist, PA_PROP_DEVICE_CLASS, "abstract");
+
+    if (pa_modargs_get_proplist(ma, "sink_properties", sink_data.proplist, PA_UPDATE_REPLACE) < 0) {
+        pa_log("Invalid properties.");
+        pa_sink_new_data_done(&sink_data);
         goto fail;
     }
 
-    u->sink = pa_sink_new(m->core, &data, 0);
-    pa_sink_new_data_done(&data);
-
-    if (!u->sink) {
+    if (!(u->sink = pa_sink_new(m->core, &sink_data, 0))) {
         pa_log("Failed to create sink object.");
+        pa_sink_new_data_done(&sink_data);
         goto fail;
     }
+    pa_sink_new_data_done(&sink_data);
 
+    u->sink->userdata = u;
     u->sink->parent.process_msg = sink_process_msg;
     u->sink->set_state_in_main_thread = state_func;
     u->sink->request_rewind = process_rewind;
-    u->sink->userdata = u;
 
     pa_sink_set_asyncmsgq(u->sink, u->thread_mq.inq);
     pa_sink_set_rtpoll(u->sink, u->rtpoll);
 
     pa_modargs_get_value_u32(ma, "latency", &latency);
-    if (latency)
-        u->block_usec = latency * PA_USEC_PER_MSEC;
-    else
-        u->block_usec = BLOCK_USEC;
+    u->block_usec = latency
+        ? PA_USEC_PER_MSEC * latency
+        : PA_USEC_PER_MSEC * 125;
     pa_sink_set_fixed_latency(u->sink, u->block_usec);
```

Comment:
I doubt that we need to / should implement PA_SINK_MESSAGE_GET_LATENCY because of this, but if you insist...